tests/prefetch_related/tests.py

```diff
@@ -1969,3 +1972,63 @@ def test_window_not_supported(self):
         )
         with self.assertRaisesMessage(NotSupportedError, msg):
             list(Book.objects.prefetch_related(Prefetch("authors", authors[1:])))
+
+
+class Ticket34791Tests(TestCase):
+    def setUp(self):
+        canada = Country.objects.create(label="Canada")
+        usa = Country.objects.create(label="USA")
+        self.china = Country.objects.create(label="China")
+        argentina = Country.objects.create(label="Argentina")
+
+        hawaian_pizza = Pizza.objects.create(label="Hawaiian", origin=canada)
+        Topping.objects.create(pizza=hawaian_pizza, label="Ham", origin=self.china)
+        Topping.objects.create(pizza=hawaian_pizza, label="Pineapple", origin=argentina)
+
+        pepperoni_pizza = Pizza.objects.create(label="Pepperoni", origin=usa)
+        Topping.objects.create(pizza=pepperoni_pizza, label="Pepperoni", origin=usa)
+
+    def test_custom_prefetch(self):
```

Comment:
A `CustomPrefetchTests` seems like a good _home_ for this test with a more appropriate name.

See `test_nested_prefetch_related_with_duplicate_prefetcher` which is slightly related.

Reply:
Makes sense, I'll move my test there and use the models already defined.